### DISCUSSION TITLE
"Apt-get upgrade" always delete my pip. What should I do?

### DISCUSSION THREAD

**Discussion by ghost** [termux/termux-packages#19242] [Q&A] [answered]
I manually installed python-pip using the following command:
```
echo "APT::Install-Suggests "0";" >../usr/etc/apt/apt.conf.d/02norecommend
echo "APT::Install-Recommends "0";" >>../usr/etc/apt/apt.conf.d/02norecommend
curl https://bootstrap.pypa.io/get-pip.py | python
```
But! When I run "apt upgrade" to update python, pip will be removed.
I tried installing pip with apt, but, I can't stand its disk usage. After that operation, 543 MB of additional disk space will be used.
What should I do?

**Comment by TomJo2000** [comment, marked as answer]
Pip on Termux is provided by the `python-pip` package.
The unpatched pypa version is not compatible with Termux.

I suspect the reason why it is using so much space is that you are missing various other dependencies.

**Comment by ghost** [comment]
Thanks.

**Reply by sk0kanik** [reply]
when you install `python-pip`, it will also install `clang`, C/C++ compiler. Because many python packages you will install using pip contains C/C++ codes.
